mapper.go

```diff
@@ -0,0 +1,308 @@
+}
+
+// Map is the implementation of Mapper[string] for UIntParser.
+func (parser *UIntParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, "")
+	return strconv.ParseUint(src, 10, 64)
+}
+
+// IntParser is a parser to int64.
+type IntParser struct {
+	ignoreChars string
+}
+
+// NewIntParser creates IntParser.
+func NewIntParser(ignoreChars string) *IntParser {
+	return &IntParser{ignoreChars: ignoreChars}
+}
+
+// Map is the implementation of Mapper[string] for IntParser.
+func (parser *IntParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, "")
+	return strconv.ParseInt(src, 10, 64)
+}
+
+// FloatParser is a parser to float64.
+type FloatParser struct {
+	ignoreChars   string
+	decSeparators string
+}
+
+// NewFloatParser creates FloatParser.
+func NewFloatParser(ignoreChars, decSeparators string) *FloatParser {
+	return &FloatParser{ignoreChars: ignoreChars, decSeparators: decSeparators}
+}
+
+// Map is the implementation of Mapper[string] for FloatParser.
+func (parser *FloatParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, parser.decSeparators)
+	return strconv.ParseFloat(src, 64)
+}
+
+// DecimalParser is a parser to decimal.Decimal.
+type DecimalParser struct {
+	ignoreChars   string
+	decSeparators string
+}
+
+// NewDecimalParser creates DecimalParser.
+func NewDecimalParser(ignoreChars, decSeparators string) *DecimalParser {
+	return &DecimalParser{ignoreChars: ignoreChars, decSeparators: decSeparators}
+}
+
+// Map is the implementation of Mapper[string] for DecimalParser.
+func (parser *DecimalParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, parser.decSeparators)
+	result, err := decimal.NewDecimalFromString(src)
+	if err == nil {
+		return *result, nil
+	}
+	return nil, err
+}
+
+// UUIDParser is a parser to UUID.
+type UUIDParser struct{}
+
+// NewUUIDParser creates UUIDParser.
+func NewUUIDParser() *UUIDParser {
+	return &UUIDParser{}
+}
+
+// Map is the implementation of Mapper[string] for UUIDParser.
+func (*UUIDParser) Map(src string) (any, error) {
+	return uuid.Parse(src)
+}
+
+// StringParser is a parser to string.
+type StringParser struct{}
+
+// NewStringParser creates StringParser.
+func NewStringParser() *StringParser {
+	return &StringParser{}
+}
+
+// Map is the implementation of Mapper[string] for StringParser.
+func (*StringParser) Map(src string) (any, error) {
+	return src, nil
+}
+
+// DateTimeParser is a parser to datetime.Datetime.
+type DateTimeParser struct{}
+
+// NewDateTimeParser creates DateTimeParser.
+func NewDateTimeParser() *DateTimeParser {
+	return &DateTimeParser{}
+}
+
+// Map is the implementation of Mapper[string] for DateTimeParser.
+func (*DateTimeParser) Map(src string) (any, error) {
+	if result, err := time.Parse(time.RFC3339, src); err == nil {
+		return datetime.NewDatetime(result)
+	}
+	return nil, fmt.Errorf("unexpected date format")
```

Comment:
It is a confusing error handling.
```suggestion
	result, err := time.Parse(time.RFC3339, src)
	if err != nil {
	    return nil, fmt.Errorf("unexpected date format: %w", err)
	}
	return result, nil
```

Reply:
Fixed.